distributed/diagnostics/cluster_dump.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+from typing import Any, Collection, Dict, Literal
+
+from distributed.diagnostics.plugin import SchedulerPlugin
+from distributed.scheduler import Scheduler
+
+
+class ClusterDump(SchedulerPlugin):
+    """Dumps cluster state prior to Scheduler shutdown
+
+    The Scheduler may shutdown in cases where it is in an error state,
+    or when it has been unexpectedly idle for long periods of time.
+    This plugin dumps the cluster state prior to Scheduler shutdown
+    for debugging purposes.
+    """
+
+    def __init__(
+        self,
+        scheduler: Scheduler,
+        url: str = None,
+        exclude: "Collection[str]" = (),
```

Comment:
Can we match this to the default exclude on the client? https://github.com/dask/distributed/blob/1b4d13fe4f6898bb825bd9e58dc3e795a7230be5/distributed/client.py#L3838

Could be reasonable to factor this out into a global tuple in `cluster_dump.py`.

Reply:
Good idea. I've add `DEFAULT_CLUSTER_DUMP_EXCLUDE` and  `DEFAULT_CLUSTER_DUMP_FORMAT` constants in `distributed.cluster_dump` and re-used them in `distributed.scheduler` and `distributed.diagnostics.plugins.cluster_dump`